crates/driver/src/tests/setup/solver.rs

```diff
@@ -168,75 +168,154 @@ impl Solver {
+                                }).collect_vec(),
+                            })
+                        }));
+                        prices_json.insert(
+                            config
+                                .blockchain
+                                .get_token_wrapped(jit.quoted_order.order.sell_token),
+                            jit.execution.buy.to_string(),
+                        );
+                        prices_json.insert(
+                            config
+                                .blockchain
+                                .get_token_wrapped(jit.quoted_order.order.buy_token),
+                            (jit.execution.sell - jit.quoted_order.order.surplus_fee()).to_string(),
+                        );
+                        {
+                            let executed_amount = match jit.quoted_order.order.executed {
+                                Some(executed) => executed.to_string(),
+                                None => match jit.quoted_order.order.side {
+                                    order::Side::Sell => (jit.execution.sell
+                                        - jit.quoted_order.order.surplus_fee())
+                                    .to_string(),
+                                    order::Side::Buy => jit.execution.buy.to_string(),
+                                },
+                            };
+                            let order = json!({
+                                "sellToken": config.blockchain.get_token(jit.quoted_order.order.sell_token),
+                                "buyToken": config.blockchain.get_token(jit.quoted_order.order.buy_token),
+                                "receiver": hex_address(jit.quoted_order.order.owner),
+                                "sellAmount": jit.quoted_order.order.sell_amount.to_string(),
+                                "buyAmount": jit.quoted_order.order.buy_amount.unwrap_or_default().to_string(),
+                                "validTo": jit.quoted_order.order.valid_to,
+                                "appData": jit.quoted_order.order.app_data,
+                                "feeAmount": jit.quoted_order.order.solver_fee.unwrap_or_default().to_string(),
+                                "kind": match jit.quoted_order.order.side {
+                                            order::Side::Sell => "sell",
+                                            order::Side::Buy => "buy",
+                                },
+                                "partiallyFillable": matches!(jit.quoted_order.order.partial, Partial::Yes { .. }),
+                                "sellTokenBalance": jit.quoted_order.order.sell_token_source,
+                                "buyTokenBalance": jit.quoted_order.order.buy_token_destination,
+                                "signature": if config.quote { "0x".to_string() } else { format!("0x{}", hex::encode(jit.quoted_order.order_signature(config.blockchain))) },
```

Comment:
This currently signs the JIT orders with the trader's private key (taken from `quoted_order`) meaning the JIT order's sell_token balance will be taken from the original trader.